### DISCUSSION TITLE
How to add Executable Code in revealjs?

### DISCUSSION THREAD

**Discussion by aborruso** [quarto-dev/quarto-cli#546] [Q&A] [answered]
Hi, quarto is really great, thank you to all developers and contributors.

I have this hello world slides

~~~
---
title: "Habits"
author: "John Doe"
format:
  revealjs:
    theme: dark
jupyter: python3
---

## Getting up

- Turn off alarm
- Get out of bed


---

```{python}
#| echo: true

import matplotlib.pyplot as plt
plt.plot([1,2,3,4])
plt.show()
```
~~~

If I open the html output the code is not executed and I have a code slide. How to have the execution output in the slide?

Thank you

![image](https://user-images.githubusercontent.com/30607/161935524-da7a93ab-ab1e-44b1-975b-25ca77391cac.png)

**Comment by cderv** [comment]
Using last release (v0.9.188), I correctly get this slide using your example code 
![image](https://user-images.githubusercontent.com/6791940/161942235-915f437b-73bb-4b1d-a524-a20982557c9c.png)

Which version of **quarto** are you using ?

**Reply by aborruso** [reply]
> Which version of **quarto** are you using ?

Thank you. I'm using 0.9.188.

I have no result both running `quarto render test.md --to revealjs` and `quarto preview test.md`

**Reply by aborruso** [reply]
The HTML I have is this

```
<section class="slide level2">

<div class="sourceCode" id="cb1"><pre class="sourceCode numberSource {python} number-lines code-with-copy"><code class="sourceCode"><span id="cb1-1"><a href="#cb1-1"></a>#| echo: true</span>
<span id="cb1-2"><a href="#cb1-2"></a></span>
<span id="cb1-3"><a href="#cb1-3"></a>import matplotlib.pyplot as plt</span>
<span id="cb1-4"><a href="#cb1-4"></a>plt.plot([1,2,3,4])</span>
<span id="cb1-5"><a href="#cb1-5"></a>plt.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="footer footer-default">

</div>
</section>
```

**Comment by jjallaire** [comment, marked as answer]
In order for code to be executed you need to use the `.qmd` extension.

**Reply by aborruso** [reply]
It works, thank you.

Is there some note about it in the documentation?

I'm starting from [this](https://quarto.org/docs/presentations/revealjs/#executable-code) and [this](https://quarto.org/docs/computations/python.html).

Some note like "Please note: the use of the extension qmd is mandatory if you want to get execution output".

**Comment by jjallaire** [comment]
One thing we tried to do was have zero examples in the docs of using `.md` (even for static content). This is mostly because the `.qmd` file extension serves as a convenient sentinel to special editor modes (e.g. in RStudio, EMacs, and VS Code) that they are dealing with a Quarto document rather than just plain markdown. Net: Use of plain `.md` is supported for compatibility but generally not encouraged even for documents without code.

**Reply by aborruso** [reply]
I'm probably stupid. But reading it, I didn't realize that the extension was mandatory.

Thank you again.

Quarto is really great

**Comment by jjallaire** [comment]
No, that's a completely natural conclusion to draw! I think the right behavior is for us to error in this case so you get feedback as to why things aren't working. I've made that change here: https://github.com/quarto-dev/quarto-cli/commit/86ae575dbfc0f457fa782b6e5f63f6925d72ebb3

**Reply by aborruso** [reply]
Great product and great community

Thank you
